python/pyspark/pandas/frame.py

```diff
@@ -6054,17 +6056,21 @@ def pivot_table(
                     # E.g. if column is b and values is ['b','e'],
                     # then ['2_b', '2_e', '3_b', '3_e'].
 
-                    # We sort the columns of Spark DataFrame by values.
-                    data_columns.sort(key=lambda x: x.split("_", 1)[1])
```

Comment:
Can we just simply use `-1` index? Calling `Series.unique` will trigger aggregation on each column that's pretty expensive. We could alternatively just subtract original column names instead of invoking `unique`.

Reply:
Thank you for the feedback, I appreciate it! :)

I tried using `-1` index, but it doesn't give the sort order expected by the tests. The `Series.unique` should only happen for a single column -- the docstring says only a single column is supported ([Link](https://github.com/apache/spark/blob/master/python/pyspark/pandas/frame.py#L5861-L5863)). Agreed that it's expensive, but I am not sure what else to do.

What do you think about using: ` _columns = [str(i) for i in set(self[columns].tolist())]` instead of `_columns = [str(i) for i in self[columns].unique().tolist()]`? Would this be less expensive?

Reply:
I went ahead and tried a slightly different approach that removes thee usage of `Series.unique`.

Reply:
I went ahead and added a slightly different approach that removes the usage of `Series.unique`.